packages/unmock-core/src/service/spy/index.ts

```diff
@@ -0,0 +1,63 @@
+import { SinonSpy, spy as sinonSpy } from "sinon";
+import { ISerializedRequest, ISerializedResponse } from "../../interfaces";
+
+export type RequestResponseSpy = SinonSpy<
+  [ISerializedRequest],
+  ISerializedResponse
+>;
+
+interface IRequestResponsePair {
+  req: ISerializedRequest;
+  res: ISerializedResponse;
+}
+
+export interface ICallTracker {
+  /**
+   * Keep track of calls, added via the `notify` method
+   */
+  spy: RequestResponseSpy;
+  /**
+   * Add new call to the spy object
+   */
+  track(pair: IRequestResponsePair): void;
+  reset(): void;
+}
+
+export type RequestResponseSpyNotifier = CallTracker<
+  ISerializedRequest,
+  ISerializedResponse
+>;
+
+/**
+ * Container for tracking spy calls via `track` method,
+ * calls are exposed via `spy` property.
+ */
+class CallTracker<TArg = any, TReturnValue = any> {
```

Comment:
`... implements ICallTracker`? (not 100% convinced an interface is needed here, especially if we'll move this to be under the `Service` class)

Reply:
Could implement if it wasn't generic. Maybe it would be cleaner to have the types be concrete but they don't really need to be so thought it could be generic.